### PR TITLE
WarpX 23.07

### PR DESCRIPTION
**Pull request by ax3l**
Update WarpX and related Python packages to the lastest releases.

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed the new version sha256, rest LGTM.

**Comment by tldahlgren** [comment]
Looks like the PR fails to build for two CI stacks due to `py-picmistandard`.

**Comment by adamjstewart** [comment]
@ax3l can you verify the checksums for picmicstandard and see where the incorrect ones are coming from? Might just be from GitHub instead of PyPI.

**Review by adamjstewart — DISMISSED** [review]
Changes look good, just need to fix checksum

**Comment by ax3l** [comment]
Checksums fixed! :tada: @tldahlgren @adamjstewart

**Review by tldahlgren — DISMISSED** [review]
Confirmed all of the affected version sha256.  Thanks!

File: var/spack/repos/builtin/packages/py-picmistandard/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-picmistandard/package.py:34]: Technically we should deprecate these before removing them

Comment (ax3l) [reply var/spack/repos/builtin/packages/py-picmistandard/package.py:34]: Technically yes. But I think it's ok here - we really neither use nor support the first versions that had these install bugs and no proper tags.

Also, I do not have time to figure out the Spack fetcher bug, that decides to pick random git tags for some of the releases the moment I have a single version in here that was specified by git.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-picmistandard/package.py:34]: You're the maintainer, so the final decision is up to you 😄

File: var/spack/repos/builtin/packages/py-picmistandard/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-picmistandard/package.py:18]:
```suggestion
    version("master", branch="master")
```

File: var/spack/repos/builtin/packages/py-picmistandard/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-picmistandard/package.py:29]: You could remove this, Spack only supports 3.7+ anyway

File: var/spack/repos/builtin/packages/py-warpx/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-warpx/package.py:90]: Wheel is already added by the base class and could be removed here

File: var/spack/repos/builtin/packages/py-warpx/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-warpx/package.py:78]: Python 3.7+ is the minimum version we support, so 3.6+ could be removed

File: var/spack/repos/builtin/packages/py-warpx/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-warpx/package.py:87]: I try to sort these from newest to oldest to match the version directive

**Review by adamjstewart — APPROVED** [review]
Some optional comments, feel free to merge if you think it's good enough

File: var/spack/repos/builtin/packages/py-picmistandard/package.py
Comment (ax3l) [reply var/spack/repos/builtin/packages/py-picmistandard/package.py:29]: Ok. Next release is 3.8+

**Comment by ax3l** [comment]
Thanks! I will merge this and add the suggestions to the 23.08 release #39407 :)

**Pull request merged**
